src/main/java/bio/terra/service/resourcemanagement/azure/AzureStorageAccountService.java

```diff
@@ -198,21 +198,25 @@ public AzureStorageAccountResource retrieveStorageAccountById(UUID storageAccoun
     return resourceDao.retrieveStorageAccountById(storageAccountId);
   }
 
-  /**
-   * Delete metadata and cloud storage account resource. Note: this runs in a transaction so if the
-   * cloud delete fails, the metadata deleted should rollback
-   */
-  @Transactional(propagation = Propagation.REQUIRED, isolation = Isolation.SERIALIZABLE)
-  public void deleteCloudStorageAccount(
+  public void deleteCloudStorageAccountAndMetadata(
       AzureStorageAccountResource storageAccountResource, String flightId) {
+    deleteCloudStorageAccount(storageAccountResource);
+    deleteCloudStorageAccountMetadata(storageAccountResource.getName(), flightId);
+  }
+
+  public void deleteCloudStorageAccount(AzureStorageAccountResource storageAccountResource) {
     BillingProfileModel profileModel =
         profileDao.getBillingProfileById(storageAccountResource.getProfileId());
+    logger.info("Deleting Azure storage account");
+    deleteCloudStorageAccount(profileModel, storageAccountResource);
+  }
+
+  public void deleteCloudStorageAccountMetadata(
+      String storageAccountResourceName, String flightId) {
```

Comment:
It would be nice to make this typed (and use `AzureStorageAccountResource`) but I guess it's called from a step where all we have is the name, so that's not possible?

Reply:
I'm not sure I follow - I think this is a unique storage account resource name, so a string?

I'm switched to just pass through the name to avoid passing through the entire storage account resource.

Reply:
IMO the main issue here is about separation of concerns. A caller shouldn't need to know what things this method needs from `AzureStorageAccountResource` to delete the metadata. It happens that only the name is needed, but maybe in the future some other state in that object will be needed. If that happens, using the object means the caller doesn't have to be changed.

Now that I look at `AzureStorageAccountResource` I see that it has a `flightId` property. If that was set, then this method could be written only accept `storageAccountResource`.